### PR TITLE
feat:ajaxAction传参处理

### PR DESCRIPTION
**Pull request by pianruijie**
### <samp>🤖 Generated by Copilot at 202b9bb</samp>

Refactored the `api` property of the `IAjaxAction` interface and the `fetcher` function in `AjaxAction.ts` to use a more consistent type.

### <samp>🤖 Generated by Copilot at 202b9bb</samp>

> _`Api` is no more, the `ApiObject` reigns_
> _Refactoring the code, breaking the chains_
> _`fetcher` and `IAjaxAction` aligned_
> _A new dawn of consistency we find_

### <samp>🤖 Generated by Copilot at 202b9bb</samp>

* Change the `api` property type of the `IAjaxAction` interface from `Api` to `ApiObject` to match the rest of the codebase and avoid confusion with the `Api` type that represents a string or a function. ([link](https://github.com/baidu/amis/pull/7234/files?diff=unified&w=0#diff-c85a51364485b453adba3a7d4d6cfc4ed76d74ffadc05dd7daf7b386e120c8b6L16-R16))
* Refactor the logic for fetching the data from the `api` property to separate the `data` property from the rest of the `api` object, to avoid mutating the original `api` object and to simplify the arguments passed to the `fetcher` function. ([link](https://github.com/baidu/amis/pull/7234/files?diff=unified&w=0#diff-c85a51364485b453adba3a7d4d6cfc4ed76d74ffadc05dd7daf7b386e120c8b6L55-R58))
* Remove the unnecessary type assertions for the `api` property as `ApiObject` in the `AjaxAction` class, to improve the readability and maintainability of the code. ([link](https://github.com/baidu/amis/pull/7234/files?diff=unified&w=0#diff-c85a51364485b453adba3a7d4d6cfc4ed76d74ffadc05dd7daf7b386e120c8b6L82-R81), [link](https://github.com/baidu/amis/pull/7234/files?diff=unified&w=0#diff-c85a51364485b453adba3a7d4d6cfc4ed76d74ffadc05dd7daf7b386e120c8b6L89-R88))

### CODE REVIEW

File: packages/amis-core/src/actions/AjaxAction.ts [hunk 1]
```diff
@@ -52,11 +52,10 @@ export class AjaxAction implements RendererAction {
 
     const env = event.context.env;
     try {
-      const result = await env.fetcher(
-        action.args?.api,
-        action.data ?? {},
-        action.args?.options ?? {}
-      );
+      const data = action.args?.api?.data ?? action.data ?? {};
+      const api = {...(action.args?.api ?? {})};
```
Comment (hsm-lv) [inline packages/amis-core/src/actions/AjaxAction.ts:56]: api可能是string

**Review by hsm-lv — CHANGES_REQUESTED** [review]

File: packages/amis-core/src/actions/AjaxAction.ts [hunk 2]
```diff
@@ -52,11 +52,19 @@ export class AjaxAction implements RendererAction {
...
-        action.data ?? {},
-        action.args?.options ?? {}
-      );
+      let result = null;
+      if (typeof action.args?.api === 'string') {
+        result = await env.fetcher(
+          action.args?.api,
+          action.data ?? {},
+          action.args?.options ?? {}
+        );
```
Comment (hsm-lv) [inline packages/amis-core/src/actions/AjaxAction.ts:61]: 压缩一下？
result = await env.fetcher(
        typeof api === 'string' ? api : omit(api.data, 'data'),
        (action.args?.api as any)?.data ?? action.data ?? {},
        action.args?.options ?? {}
      );

**Review by hsm-lv — CHANGES_REQUESTED** [review]

**Review by hsm-lv — APPROVED** [review]

**Comment by 2betop** [comment]
感觉应该是把 api 从 args 拿出去，因为在 args 里面会做一层基于event.data 的数据映射，而不是 action.data 的。

```
"onEvent": {
        "click": {
          "actions": [
            {
              "actionType": "ajax",
              "args": {
                "api": {
                  "url": "https://3xsw4ap8wah59.cfc-execute.bj.baidubce.com/api/amis-mock/mock2/form/saveForm?name=${name}",
                  "method": "get",
                  "responseData": {
                    "resId": "${id}"
                  }
                },
                "messages": {
                  "success": "成功了！欧耶",
                  "failed": "失败了呢。。"
                }
              },
              "data": {
                "age": 18
              }
            },
            {
              "actionType": "toast",
              "expression": "${event.data.responseResult.responseStatus === 0}",
              "args": {
                "msg": "${event.data.responseResult|json}"
              }
            }
          ]
        }
      }
```

像这个 responseData 的处理应该是接口返回后做映射，因为 api 在 args 里面在事件动作执行前做一次数据映射，到时实际拿得的数据是发送前的，而不是接口返回后的。

File: packages/amis-core/src/actions/AjaxAction.ts [hunk 3]
```diff
@@ -53,8 +54,10 @@ export class AjaxAction implements RendererAction {
     const env = event.context.env;
     try {
       const result = await env.fetcher(
-        action.args?.api,
-        action.data ?? {},
+        typeof action.args?.api === 'string'
+          ? action.args.api
```
Comment (2betop) [inline packages/amis-core/src/actions/AjaxAction.ts:58]: 应该把 api 从 args 里面移出去，目前的问题其实是 args.api.data 做了两次数据映射

**Review by hsm-lv — CHANGES_REQUESTED** [review]
待定

**Review by 2betop — APPROVED** [review]

File: packages/amis-editor/src/renderer/event-control/comp-action-select.tsx [hunk 4]
```diff
@@ -12,8 +12,8 @@ export const BASE_ACTION_PROPS = [
   '__actionDesc',
   'preventDefault',
   'stopPropagation',
-  'expression',
-  'outputVar'
+  'expression'
+  // 'outputVar'
```
Comment (hsm-lv) [inline packages/amis-editor/src/renderer/event-control/comp-action-select.tsx:16]: 这个为啥

**Review by hsm-lv — APPROVED** [review]

File: packages/amis-core/src/actions/AjaxAction.ts [hunk 5]
```diff
@@ -12,15 +12,13 @@ import {
...
   action: 'ajax';
-  args: {
-    api: Api;
-    messages?: {
-      success: string;
-      failed: string;
-    };
-    options?: Record<string, any>;
-    [propName: string]: any;
+  api: Api;
```
Comment (hsm-lv) [inline packages/amis-core/src/actions/AjaxAction.ts:15]: 文档&示例更新

**Review by hsm-lv — APPROVED** [review]

**Pull request merged**